assemble.go

```diff
@@ -137,7 +189,15 @@ func AssembleFile(ctx context.Context, name string, idx Index, s Store, seeds []
 						}
 						sum := Digest.Sum(b)
 						if sum != c.ID {
-							return fmt.Errorf("written data in %s doesn't match its expected hash value, seed may have changed during processing", name)
+							if options.InvalidSeedAction == InvalidSeedActionRegenerate {
+								// Try harder before giving up and aborting
+								Log.WithField("ID", c.ID).Info("The seed may have changed during processing, trying to take the chunk from the self seed or the store")
+								if err := writeChunk(c, ss, f, blocksize, s, stats, isBlank); err != nil {
+									return err
+								}
```

Comment:
This could be further optimized, for example by remembering which chunks were invalid and avoiding to rewriting them.
I didn't do it for now because it would increase the complexity of this PR and I wanted to start with something simpler and maybe add on top of it later on.

Reply:
Given that this only happens in (hopefully rare) situations it's perfectly fine to not worry about efficiency. No need to worry about efficiency in the failure-case.

Reply:
Right, I totally agree.